source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/CustomTextBlockRenderer.mm

```diff
@@ -0,0 +1,109 @@
+//
+//  CustomTextBlockRenderer
+//  CustomTextBlockRenderer.mm
+//
+//  Copyright © 2018 Microsoft. All rights reserved.
+//
+
+#import "CustomTextBlockRenderer.h"
+#import <AdaptiveCards/ACFramework.h>
+#import <AdaptiveCards/SharedAdaptiveCard.h>
+#import <AdaptiveCards/TextBlock.h>
+#import <AdaptiveCards/ACRTextBlockRenderer.h>
+#import <AdaptiveCards/ACOBaseCardElementPrivate.h>
+#import <AdaptiveCards/ACRContentHoldingUIView.h>
+#import <AdaptiveCards/MarkDownParser.h>
+#import <AdaptiveCards/HostConfig.h>
+#import <AdaptiveCards/ACOHostConfigPrivate.h>
+
+@implementation CustomTextBlockRenderer
+
++ (CustomTextBlockRenderer *)getInstance
+{
+    static CustomTextBlockRenderer *singletonInstance = [[self alloc] init];
+    return singletonInstance;
+}
+
+- (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
+            rootView:(ACRView *)rootView
+            inputs:(NSArray *)inputs
+   baseCardElement:(ACOBaseCardElement *)acoElem
+        hostConfig:(ACOHostConfig *)acoConfig
+{
+
+    std::shared_ptr<BaseCardElement> elem = [acoElem element];
+    std::shared_ptr<TextBlock> textBlockElement = std::static_pointer_cast<TextBlock>(elem);
+
+    struct TextConfig textConfigForBlock =
+    {
+        .weight = textBlockElement->GetTextWeight(),
+        .size = textBlockElement->GetTextSize(),
+        .color = textBlockElement->GetTextColor(),
+        .isSubtle = textBlockElement->GetIsSubtle(),
+        .wrap = textBlockElement->GetWrap()
+    };
+
+    std::string textForBlock = textBlockElement->GetText();
+
+    NSString* parsedString = nil;
+    // MarkDownParser transforms text with MarkDown to a html string
+    std::shared_ptr<MarkDownParser> markDownParser = std::make_shared<MarkDownParser>([ACOHostConfig getLocalizedDate:textBlockElement].c_str());
```

Comment:
>.c_str() [](start = 135, length = 8)

Don't call `c_str()` here. It just results in the string getting duplicated unnecessarily.

Reply:
You're still calling c_str() and creating an unnecessary copy of the string.

---
In reply to: [189051401](https://github.com/Microsoft/AdaptiveCards/pull/1445#discussion_r189051401) [](ancestors = 189051401)

Reply:
thanks!

---
In reply to: [189127697](https://github.com/Microsoft/AdaptiveCards/pull/1445#discussion_r189127697) [](ancestors = 189127697,189051401)